File: src/write.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: ./write_file <num_lines>\n");
        return 1;
    }

    int num_lines = atoi(argv[1]);
    FILE *fp = fopen("../data/test_c.txt", "w");

    if (fp == NULL) {
        printf("Error opening file.\n");
        return 1;
    }

    clock_t start_time = clock();

    for (int i = 0; i < num_lines; i++) {
        fprintf(fp, "This is line %d\n", i);
    }

    clock_t end_time = clock();

    fclose(fp);

    double run_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;

    printf("Wrote %d lines to ../data/test_c.txt\n", num_lines);
    printf("Internal runtime: %.6f seconds\n", run_time);

    return 0;
}

File: week2/src/write.py
import sys
import time

if len(sys.argv) != 2:
    print("Usage: python3 write_file.py <num_lines>")
    sys.exit(1)

num_lines = int(sys.argv[1])
filename = "../data/test_python.txt"

start_time = time.time()

with open(filename, "w") as f:
    for i in range(num_lines):
        f.write(f"This is line {i}\n")

end_time = time.time()

run_time = end_time - start_time
print(f"Wrote {num_lines} lines to {filename}")
print(f"Internal runtime: {run_time:.6f} seconds")
